clients/src/main/java/org/apache/kafka/clients/producer/RoundRobinPartitioner.java

```diff
@@ -73,4 +73,12 @@ private int nextValue(String topic) {
 
     public void close() {}
 
+    @Override
+    public void onNewBatch(String topic, Cluster cluster, int prevPartition) {
+        synchronized (topicCounterMap) {
```

Comment:
Could it be replaced by ```computeIfPresent```?

Reply:
@chia7712 Sure, will add a test for this and will try to replace this with computeIfPresent.

Reply:
@chia7712 I have replaced it with `computeIfPresent`.

Reply:
the ```synchronized``` block is redundant, right?